src/improvements/template/DisableLivenessModule2.sol

```diff
@@ -0,0 +1,187 @@
+
+    /// @notice Safe address string identifier
+    string _safeAddressString;
+
+    /// @notice Gnosis Safe Sentinel Module address
+    address internal constant SENTINEL_MODULE = address(0x1);
+
+    /// @notice Gnosis Safe Module Mapping Storage Offset
+    uint256 public constant MODULE_MAPPING_STORAGE_OFFSET = 1;
+
+    /// @notice Gnosis Safe Module Nonce Storage Offset
+    bytes32 public constant NONCE_STORAGE_OFFSET = bytes32(uint256(5));
+
+    /// @notice Returns the safe address string identifier
+    function safeAddressString() public view override returns (string memory) {
+        return _safeAddressString;
+    }
+
+    /// @notice Returns the storage write permissions required for this task
+    /// @return Array of storage write permissions
+    function _taskStorageWrites() internal pure override returns (string[] memory) {
+        // Storage writes for the module are handled in _setAllowedStorageAccesses
+        // since the module address is not in the SimpleAddressRegistry
+        return new string[](0);
+    }
+
+    /// @notice Override to add the module address to allowed storage accesses
+    function _setAllowedStorageAccesses() internal override {
+        super._setAllowedStorageAccesses();
+        // Add the module address directly since it's not in the registry
+        _allowedStorageAccesses.add(moduleToDisable);
+    }
+
+    /// @notice Sets up the template with module configuration from a TOML file
+    function _templateSetup(string memory _taskConfigFilePath, address _rootSafe) internal override {
+        super._templateSetup(_taskConfigFilePath, _rootSafe);
+        string memory file = vm.readFile(_taskConfigFilePath);
+        moduleToDisable = vm.parseTomlAddress(file, ".moduleToDisable");
+        previousModule = vm.parseTomlAddress(file, ".previousModule");
+    }
+
+    /// @notice Builds the action for disabling the module in the Safe and clearing its configuration
+    function _build(address _rootSafe) internal override {
+        // First disable the module from the Safe
+        ModuleManager(_rootSafe).disableModule(previousModule, moduleToDisable);
+
+        // Then clear the module configuration
+        ILivenessModule2(moduleToDisable).clear();
+    }
+
+    /// @notice Validates that the module was disabled and cleared correctly.
+    function _validate(VmSafe.AccountAccess[] memory _accountAccesses, Action[] memory, address _rootSafe)
+        internal
+        view
+        override
+    {
+        _validateModuleDisabled(_rootSafe);
+        _validateModuleCleared(_rootSafe);
+        _validateStorageWrites(_accountAccesses, _rootSafe);
+    }
+
+    function _validateModuleDisabled(address _rootSafe) internal view {
+        (address[] memory modules, address nextModule) =
+            ModuleManager(_rootSafe).getModulesPaginated(SENTINEL_MODULE, 100);
+        if (keccak256(abi.encodePacked(ISafe(_rootSafe).VERSION())) != keccak256(abi.encodePacked("1.1.1"))) {
```

Comment:
This `if` can be the first line in the function.

Also, what's special about version `1.1.1` that we don't check `isModuleEnabled` for it?

Reply:
isModuleEnabled does not exist on the 1.1.1 interface, this is what we use elsewhere on superchain-ops.  That's why we use the loops